docs/concepts/submit-data.md

```diff
@@ -0,0 +1,231 @@
+how many shares are needed to store the blob size. Then, it computes the product
+of the number of shares, the number of bytes per share, and the `gasPerByte`
+parameter. Finally, it adds a static amount per blob.
+
+The gas cost per blob byte and gas cost per transaction byte are parameters that
+could potentially be adjusted through the system's governance mechanisms. Hence,
+actual costs may vary depending on the current settings of these parameters.
+
+#### Estimating Gas Programmatically
+
+Users can estimate an efficient gas limit by using this function:
+
+```go
+import (
+    blobtypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+gasLimit := blobtypes.DefaultEstimateGas([]uint32{uint32(sizeOfDataInBytes)})
+```
+
+If using a celestia-node light client, then this function is automatically
+called for you when submitting a blob. This function works by breaking down the
+components of calculating gas for a blob transaction. These components consist
+of a flat costs for all PFBs, the size of each blob and how many shares each
+uses and the parameter for gas used per byte. More information about how gas is
+used can be found in the [gas
+specs](https://github.com/celestiaorg/celestia-app/blob/d17e231ae3a0150b50a1854f3e9a268c34502b6b/specs/src/specs/resource_pricing.md)
+and the exact formula can be found in the [blob
+module](https://github.com/celestiaorg/celestia-app/blob/d17e231ae3a0150b50a1854f3e9a268c34502b6b/x/blob/types/payforblob.go#L157-L181).
+
+### Submitting Multiple Transactions in One Block From The Same Account
+
+The mempool Celestia uses works by maintaining a fork of the canonical state
+each block. This means that each time we submit a transaction to it, it will
+update the sequence number (aka nonce) for the account that submitted the
+transaction. If users wish to submit a second transaction, they can, but must
+specify the nonce manually. If this is not done, the new transactions will not
+be able to be submitted until the first transactions is pruned.
+
+By default, nodes will prune a transaction if it does not get included in 10
+blocks (roughly 2.5 minutes). At this point, the user must resubmit their
+transaction if they want it to eventually be included.
+
+As of v1.0.0 of the application (celestia-app), users are unable to replace an
+existing transaction with a different one with higher fees. They must instead
+wait 10 blocks from the original submitted time and then resubmit the
+transaction. Again, community members have already suggested solutions and a
+willingness to accept a changes to fix this issue.
+
+## API
+
+Users can currently create and submit BlobTxs in four ways.
+
+### The celestia-app consensus node CLI
+
+```terminal
+celestia-appd tx blob PayForBlobs <hex encoded namespace> <hex encoded data> [flags]
+```
+
+### The celestia-node light node CLI
+
+```terminal
+celestia rpc blob submit <namespace in hex> <data in hex> [flags]
+```
```

Comment:
Additionally document the `state` alternative, which allows setting the gas limit and fee.

Reply:
great idea. i'll also add this to node-tutorial in this PR.

Reply:
thanks for handling this @jcstein 

for my own curiosity, where can we find info on the state alternative? I compiled main of celestia-node but only see `blob` as a subcommand

Reply:
you're welcome, thanks for making the PR

here is the link to the latest spec in api docs: https://docs.celestia.org/api/v0.11.0-rc11/#state.SubmitPayForBlob

but i am currently having trouble using it through the cli, in this [slack thread](https://celestia-team.slack.com/archives/C02F3DUTKEK/p1694017022600569?thread_ts=1693253744.690649&cid=C02F3DUTKEK)